tests/test_ssa_value.py

```diff
@@ -59,3 +60,16 @@ def test_invalid_ssa_vals(name: str):
     val = BlockArgument(i32, Block(), 0)
     with pytest.raises(ValueError):
         val.name_hint = name
+
+
+def test_typed_ssavalue():
+    """
+    Check that SSAValue type hinting works as expected.
+    This is a type hinting test.
+    """
+    value = TestSSAValue(i32)
+
+    def foo(arg: SSAValue[IntegerType]):
+        pass
+
+    foo(value)
```

Comment:
is there any way we can make a test that shows a type error? I guess not as it would have to rely on pyright

Reply:
Yeah I'm discussing it with Sasha, we are trying to find a solution

Reply:
I don't think we need this?

Reply:
removed